clang/lib/CConv/AVarBoundsInfo.cpp

```diff
@@ -530,11 +533,6 @@ void AVarBoundsInfo::insertParamKey(AVarBoundsInfo::ParamDeclType ParamDecl,
 }
 
 void AVarBoundsInfo::insertProgramVar(BoundsKey NK, ProgramVar *PV) {
-  if (getProgramVar(NK) != nullptr) {
-    // Free the already created variable.
-    auto *E = PVarInfo[NK];
-    delete (E);
-  }
```

Comment:
Why isn't the fix just to delete this code? I.e., why do you also need to create the set of vars?

Reply:
If we just delete this code, without storing the `ProgramVar *` elsewhere We will lose pointers to previously allocated `ProgramVar` objects. So, in the future, lets say we want to free all unused memory, it may not be possible.

Reply:
OK, I see.